components/Editor/index.tsx

```diff
@@ -119,9 +127,46 @@ const Editor = ({ readOnly = false }: Props) => {
     return _callValue
   }, [callValue, unit])
 
+  const deployByteCode = useCallback(
+    (bc, args = '', callValue) => {
+      try {
+        if (!callValue) {
+          callValue = getCallValue()
+        }
+        return transactionData(bc + args, callValue).then((tx) => {
+          loadInstructions(bc)
+          setIsCompiling(false)
+          return startTransaction(tx).then((result) => {
```

Comment:
probably best not to nest this promise in here like this -- we could either chain it and then explicitly handle a `.catch`, or we could just convert this whole outer function to be `async`, e.g.

```typescript
const deployByteCode = useCallback(
  async (bc: string, args: string, callValue: BN | undefined) => {
    try {
      if (!callValue) {
        callValue = getCallValue()
      }
      const tx = await transactionData(bc + args, callValue)
      loadInstructions(bc)
      setIsCompiling(false)
      const result = await startTransaction(tx)

      if (
        codeType === CodeType.Solidity &&
        !result.error &&
        result.returnValue
      ) {
        setMethodByteCode(bufferToHex(result.returnValue))
      }
      return result
    } catch (error) {
      log((error as Error).message, 'error')
      setIsCompiling(false)
    }
  },
  [
    transactionData,
    getCallValue,
    loadInstructions,
    startTransaction,
    codeType,
    log,
  ],
)
```
this way we `catch` any errors from all promises in the `try` block :)

Reply:
Ok, I will use ```async``` and ```await``` to change the ```promise```